modules/theme-tools/content-options/featured-images.php

```diff
@@ -57,3 +58,22 @@ function jetpack_featured_images_remove_post_thumbnail( $metadata, $object_id, $
 function jetpack_is_product() {
 	return ( function_exists( 'is_product' ) ) ? is_product() : false;
 }
+
+/**
+ * Check if we are on a WooCommerce Shop in order to exclude it from the is_archive check.
+ */
+function jetpack_is_shop_page() {
+	global $wp_query;
+
+	$front_page_id        = get_option( 'page_on_front' );
+	$current_page_id      = $wp_query->get( 'page_id' );
+	$is_static_front_page = 'page' === get_option( 'show_on_front' );
+
+	if ( $is_static_front_page && $front_page_id === $current_page_id  ) {
+		$is_shop_page = ( $current_page_id === wc_get_page_id( 'shop' ) ) ? true : false;
```

Comment:
Should we check that `wc_get_page_id` exists before to use it?

Reply:
You're totally right! There's another WooCommerce function in there too - `is_shop()`. I'll update shortly.